libtextsecure/message_receiver.js

```diff
@@ -1048,6 +1058,32 @@ MessageReceiver.prototype.extend({
     }
     return this.removeFromCache(envelope);
   },
+  async updateProfile(conversation, profile) {
+    // Retain old values unless changed:
+    const newProfile = conversation.get('profile') || {};
+
+    newProfile.displayName = profile.displayName;
+
+    // TODO: may need to allow users to reset their avatars to null
+    if (profile.avatar) {
+      const prevPointer = conversation.get('avatarPointer');
+      const needsUpdate =
+        !prevPointer ||
+        (prevPointer && !_.isEqual(prevPointer, profile.avatar));
```

Comment:
Left half of this && is implied by the ||

Reply:
updated